### PR TITLE
Added support for atleast_1d, atleast_2d, atleast_3d

### PR DESCRIPTION
**Pull request by hinriksnaer**
## Proposed changes

Feature addition for the following [issue](https://github.com/ml-explore/mlx/issues/668). The code is based on the numpy [implementation](https://github.com/numpy/numpy/blob/v1.26.0/numpy/core/shape_base.py#L81-L132). PyTorch provides [similar functionality](https://github.com/pytorch/pytorch/blob/3470ab42bb754a973e27f552faf0f800edea1d14/aten/src/ATen/native/TensorTransformations.cpp#L170) but is limited to only Tensors and implements it in the C++ backend. Perhaps this is a good place to start?

Any feedback regarding the documentation, tests and implementation would be much appreciated.

## Checklist

Put an `x` in the boxes that apply.

- [x] I have read the [CONTRIBUTING](https://github.com/ml-explore/mlx/blob/main/CONTRIBUTING.md) document
- [x] I have run `pre-commit run --all-files` to format my code / installed pre-commit prior to committing changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation (if needed)

### CODE REVIEW

File: docs/src/python/atleast_utils.rst [hunk 1]
```diff
@@ -0,0 +1,15 @@
+.. _utils:
+
+At least Utils
```
Comment (awni) [inline docs/src/python/atleast_utils.rst:3]: Don't need a separate page in the docs for this, I would just put them in the `ops.rst` list.

File: python/mlx/utils.py [hunk 2]
```diff
@@ -143,3 +145,103 @@ def tree_unflatten(tree):
         return l
     else:
         return {k: tree_unflatten(v) for k, v in children.items()}
+
+
+def atleast_1d(*values):
```
Comment (awni) [inline python/mlx/utils.py:150]: These should be implemented in C++ not Python and then bound to Python via Pybind (the same as all the rest of our ops).

**Review by awni — CHANGES_REQUESTED** [review]
These should follow the implementation of all of our other ops (C++ and then bound to Python). Are you up for doing the C++ implementation?

**Comment by hinriksnaer** [comment]
Thanks for the feedback! Yes, I'll get started on implementing the functionality using C++. Just to be sure, I presume that you want the functionality to be added [here](https://github.com/ml-explore/mlx/blob/main/mlx/ops.cpp), correct? 

Please let me know if there is anything else you would want me to know before getting started.

**Comment by awni** [comment]
That's exactly right!

**Pull request closed without merging**